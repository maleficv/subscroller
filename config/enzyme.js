import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import fetch from 'isomorphic-fetch';

global.fetch = fetch;

Enzyme.configure({adapter: new Adapter()});