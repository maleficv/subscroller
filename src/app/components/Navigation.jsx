import React, {Component} from 'react';
import styled from 'react-emotion';
import {MdMenu, MdClose} from 'react-icons/md';
import colors from '../theme';

import NavLinks from './NavLinks';

const Nav = styled('nav')`
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba(0, 0, 0, 0.9);
  display: flex;
  box-sizing: border-box;
  height: 100%;
  padding: 10px;
  text-align: left;
  flex-direction: column;
  justify-content: space-between;
  z-index: 3;
  color: white;
  font-size: 16px;

  transform: translateX(${props => props.active ? '0%' : '-100%'});
  transition: transform 0.2s ease;
`;

const Toggle = styled('button')`
  position: absolute;
  top: 40px;
  right: 0;
  display: flex;
  transform: translateX(100%);
  padding: 12px;
  background-color: ${colors.primary};
  border: 0;
  border-left: 0px;
  font-size: 24px;
  
  &::-moz-focus-inner {
    border: 0;
  }
`;

const Divider = styled('span')`
  width: 100%;
  height: 1px;
  margin-top: 20px;
  margin-bottom: 20px;
  background-color: white;
`;

const Config = styled('div')`
  color: white;
`;

const Option = styled('div')`
  display: flex;
  justify-content: space-between;
`;

class Navigation extends Component {

    constructor(props) {
        super(props);
        this.state = {
            active: false,
            nsfw: localStorage.getItem('nsfw') === 'true',
            grid: 3
        };

        this.toggleNavigation = this.toggleNavigation.bind(this);
        this.toggleNSFW = this.toggleNSFW.bind(this);
    }

    toggleNavigation() {
        this.setState(() => {
            return {
                active: !this.state.active
            }
        })
    }

    toggleNSFW(event) {
        event.persist();
        this.setState(() => {
            return {
                nsfw: event.target.checked
            }
        }, () => {
            localStorage.setItem('nsfw', event.target.checked.toString());
        })
    }

    render() {
        return (
            <Nav className="navigation" active={this.state.active}>
                <NavLinks toggleNavigation={this.toggleNavigation} nsfw={this.state.nsfw}/>
                <Config>
                    <Divider/>
                    <Option>Show NSFW content
                        <input type="checkbox"
                               checked={this.state.nsfw}
                               onChange={this.toggleNSFW}/>
                    </Option>
                </Config>

                <Toggle active={this.state.active} onClick={this.toggleNavigation}>
                    {this.state.active ? <MdClose/> : <MdMenu/>}
                </Toggle>
            </Nav>
        )
    }
}

export default Navigation;