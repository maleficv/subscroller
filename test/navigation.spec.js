import React from 'react';

import {shallow} from 'enzyme';

import Navigation from '../src/app/components/Navigation';

describe('Navigation component', () => {
    const nav = shallow(<Navigation/>);

    test('Navigation matches snapshot', () => {
        expect(nav).toMatchSnapshot();
    });
});