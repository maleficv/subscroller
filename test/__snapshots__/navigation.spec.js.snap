// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Navigation matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "active": false,
      "children": Array [
        <NavLinks
          nsfw={false}
          toggleNavigation={[Function]}
        />,
        <Styled(div)>
          <Styled(span) />
          <Styled(div)>
            Show NSFW content
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </Styled(div)>
        </Styled(div)>,
        <Styled(button)
          active={false}
          onClick={[Function]}
        >
          <MdMenu />
        </Styled(button)>,
      ],
      "className": "navigation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "nsfw": false,
          "toggleNavigation": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(span) />,
            <Styled(div)>
              Show NSFW content
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
            </Styled(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Show NSFW content",
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Show NSFW content",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": <MdMenu />,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "active": false,
        "children": Array [
          <NavLinks
            nsfw={false}
            toggleNavigation={[Function]}
          />,
          <Styled(div)>
            <Styled(span) />
            <Styled(div)>
              Show NSFW content
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />
            </Styled(div)>
          </Styled(div)>,
          <Styled(button)
            active={false}
            onClick={[Function]}
          >
            <MdMenu />
          </Styled(button)>,
        ],
        "className": "navigation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "nsfw": false,
            "toggleNavigation": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(span) />,
              <Styled(div)>
                Show NSFW content
                <input
                  checked={false}
                  onChange={[Function]}
                  type="checkbox"
                />
              </Styled(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Show NSFW content",
                  <input
                    checked={false}
                    onChange={[Function]}
                    type="checkbox"
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Show NSFW content",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "checked": false,
                    "onChange": [Function],
                    "type": "checkbox",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": <MdMenu />,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
